text-serializer-ansi/src/main/java/net/kyori/adventure/text/serializer/ansi/AnsiComponentSerializerImpl.java

```diff
@@ -0,0 +1,273 @@
+ *
+ * Copyright (c) 2017-2020 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.text.serializer.ansi;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.TextComponent;
+import net.kyori.adventure.text.TranslatableComponent;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.text.format.TextColor;
+import net.kyori.adventure.text.format.TextDecoration;
+import net.kyori.adventure.text.format.TextFormat;
+import net.kyori.adventure.text.renderer.TranslatableComponentRenderer;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.fusesource.jansi.Ansi;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class AnsiComponentSerializerImpl implements AnsiComponentSerializer {
+
+  private final boolean colorDownSample;
+
+  @SuppressWarnings({"WeakerAccess"})
+  public static AnsiComponentSerializer trueColor = new AnsiComponentSerializerImpl(false);
+
+  private AnsiComponentSerializerImpl(final boolean colorDownSample) {
+    this.colorDownSample = colorDownSample;
+  }
+
+  private static final TextDecoration[] DECORATIONS = TextDecoration.values();
+  protected static final String RGB_FORMAT = ESC_CHAR + "[38;2;%d;%d;%dm";
+  private static final Pattern STRIP_ESC_CODES = Pattern.compile(ESC_CHAR + "\\[[;\\d]*m");
+  private static final List<TextFormat> FORMATS;
+
+  static {
+    final List<TextFormat> formats = new ArrayList<>(NamedTextColor.values());
+    Collections.addAll(formats, DECORATIONS);
+    formats.add(Reset.INSTANCE);
+    FORMATS = Collections.unmodifiableList(formats);
+
```

Comment:
```suggestion
```

Reply:
actually the whole static constructor isnt required at the moment - I used those collections to actually do the deserialization...which we don't need.